redis-om-spring/src/main/java/com/redis/om/spring/RedisEnhancedKeyValueAdapter.java

```diff
@@ -236,6 +239,39 @@ public <T> T get(Object id, String keyspace, Class<T> type) {
     return readBackTimeToLiveIfSet(binId, converter.read(type, data));
   }
 
+  /*
+   * (non-Javadoc)
+   * 
+   * @see
+   * org.springframework.data.keyvalue.core.AbstractKeyValueAdapter#delete(java.
+   * lang.Object, java.lang.String, java.lang.Class)
+   */
+  @Override
+  public <T> T delete(Object id, String keyspace, Class<T> type) {
+    T o = get(id, keyspace, type);
+
+    if (o != null) {
+
+      byte[] keyToDelete = createKey(asString(keyspace), asString(id));
+
+      redisOperations.execute((RedisCallback<Void>) connection -> {
+
+        connection.del(keyToDelete);
+
+        if (keepShadowCopy()) {
+          RedisPersistentEntity<?> persistentEntity = converter.getMappingContext().getPersistentEntity(type);
+          if (persistentEntity != null && persistentEntity.isExpiring()) {
```

Comment:
is there a bit of context here I'm missing? The boolean for `persistentEntity.isExpiring()` seems inverted. Unless the intention is to persist the ghosts of expired objects as they die?